### ISSUE TITLE
Type 'spline' changed to 'line' causes problems going to other types after that.

### ISSUE THREAD

**Issue by jfallon1** [highcharts/highcharts#2322]
I am using 3.0.6. The specific issue noted in Issue #2270 has been fixed. However, there are other similar problems.

I have 3 series in a chart of type=spline. I can toggle between line and spline correctly. However, when I go from Line to Area I get an AreaSpline instead. And when I click Line I get a Spline instead. Clicking Line after that does nothing. But if I click Spline and THEN click Line the chart is re-drawn correctly.

Similarly, if I click Column it renders correctly. But when I click Line after that it renders a Spline instead.

Same thing for Scatter and Pie charts. 

Bottom line - if the chart is first set up as a Spline then you click a button to change it to Line, the renderings of other buttons do not work correctly going forward or coming back to Line.

On the positive side: if the chart starts life as a Line chart then all transitions to other types seem to work correctly.

-Joe

**Comment by TorsteinHonsi** [comment]
Here's a new demo, probably of the same issue: http://jsfiddle.net/highcharts/97651yvn/5/. The console log shows how a prototype property from Candlestick is first removed, then it appears again on the second time the button is clicked.
